models/DataObject/AbstractObject.php

```diff
@@ -1230,11 +1231,7 @@ public function getParent()
      */
     public function setParent($o_parent)
     {
-        $newParentId = $o_parent instanceof self ? $o_parent->getId() : 0;
-        $this->setParentId($newParentId);
-        $this->o_parent = $o_parent;
-
-        return $this;
+        return parent::setParent($o_parent);
     }
```

Comment:
Shouldn't it be enough to annotate the class with `@method $this setParent(self|null $parent)`?

Reply:
not, when we start using php type hints, right?